bundles/nl.asml.matala.product/src/nl/asml/matala/product/validation/ProductValidator.xtend

```diff
@@ -30,6 +39,48 @@ class ProductValidator extends AbstractProductValidator {
     }
     
     
+    /**
+     * Check the duplication of variables in Inputs
+     */
+     
+     @Check
+     def checkInputDuplication(Block b){ 
+            var HashSet<String> existingInputs = new HashSet<String>();
+            for (input : b.invars) {
+                if (!existingInputs.add(input.name)) {
+                    error("There exists a variable name in inputs with the same name" ,  ProductPackage.Literals.BLOCK__INVARS)  
```

Comment:
Currently the first occurrence with the name will not get the error, but it could be the erroneous one. So report this problem on all duplicate occurrences.
Aside that, the error API also allows you to write the error on the n-th index of a containment relation. This will result in better error reporting to the end user, so please use that.